### PR TITLE
Text Value as Html

### LINKED ISSUE #6: Viewer Value Component: TextValueAsHTMLComponent
Implement BaseValueComponent for `TextValue` for a `TextValueAsHtml`.

There is only read mode.

### PR DESCRIPTION
**Pull request by tobiasschweizer**
replaces #50

closes #6

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@mdelez I think I did what's necessary for text value as html. I am waiting for the generated test data PR, but I think you can already have a look at this PR whenever you have time.

File: projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div [innerHTML]="htmlFromKnora" class="value"></div>
```
Comment (mdelez) [inline projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html:1]: The html is okay for now, but the CSS I've been working on will need to be adapted to fit this readonly, non form value

File: projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
     <kui-link-value #displayVal *ngSwitchCase="constants.LinkValue" [mode]="mode" [displayValue]="displayValue"
                     [parentResource]="parentResource" [propIri]="displayValue.property"></kui-link-value>
   </span>
 </div>
 <div>
-  <button class="edit" *ngIf="canModify && !editModeActive" (click)="activateEditMode()">edit</button>
-  <button class="save" *ngIf="editModeActive" [disabled]="!displayValueComponent.form.valid"
-    (click)="saveEditValue()">Save</button>
-  <button class="cancel" *ngIf="editModeActive" (click)="cancelEditValue()">Cancel</button>
+  <span *ngIf="valueTypeOrClass !== 'ReadTextValueAsHtml'">
```
Comment (mdelez) [inline projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html:17]: maybe change this to check for a "readonly" boolean (ex: readonlyProp) in the event that we have more readonly properties in the future which can be set in the `getValueTypeOrClass` method

File: projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    });
+
+    it('should display an existing value with a comment', () => {
+
+      const inputVal: ReadTextValueAsHtml = new ReadTextValueAsHtml();
+
+      inputVal.type = 'http://api.knora.org/ontology/knora-api/v2#TextValue';
+      inputVal.id = 'http://rdfh.ch/0001/H6gBWUuJSuuO-CilHV8kQw/values/TEST_ID';
+      inputVal.html =
+        '<p>This is a <b>very</b> simple HTML document with a <a href="https://www.google.ch" target="_blank" class="kui-link">link and a comment</a></p>';
```
Comment (mdelez) [inline projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts:108]: Small change, just move the text "and a comment" outside of the anchor tag and update the expected values below

File: projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html (hunk 2 above)
Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html:17]: So you suggest to evaluate this in a method and then just check for a boolean in the template? I think that's a good idea, will do.

File: projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts (hunk 3 above)
Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts:108]: Absolutely!

File: projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html (hunk 2 above)
Comment (mdelez) [reply projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html:17]: exactly!

Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.html]: done in https://github.com/dasch-swiss/knora-ui-ng-lib/pull/55/commits/a3d215f8f7442b1414b3af0d2c563b97f338a68b

File: projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts (hunk 3 above)
Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts]: done in https://github.com/dasch-swiss/knora-ui-ng-lib/pull/55/commits/7db19f643d18b199f0268890b92fcd14c4ba6bd3

File: projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html (hunk 1 above)
Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html:1]: Shall we merge this PR first so you can integrate it into #43?

Comment (mdelez) [reply projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html:1]: yeah I think that sounds good!

Comment (tobiasschweizer) [reply projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.html:1]: ok, great!

**Pull request merged**